src/jsc/jsc_function.hpp

```diff
@@ -29,6 +29,17 @@ inline JSValueRef jsc::Function::call(JSContextRef ctx, const JSObjectRef& funct
 {
     JSValueRef exception = nullptr;
     JSValueRef result = JSObjectCallAsFunction(ctx, function, this_object, argc, arguments, &exception);
+
+    // Flush the React Native UI task queue whenever we call into JS from C++ – see `_flushUiTaskQueue` in
+    // `lib/extensions.js` for detailed explanation of why this is necessary.
+    JSObjectRef global_object = JSContextGetGlobalObject(ctx);
+    JSObjectRef realm_constructor = jsc::Object::validated_get_constructor(ctx, global_object, "Realm");
+    JSObjectRef flush_ui_task_queue =
```

Comment:
I think a check for this being `undefined` before calling it would be good.

Reply:
Yeah, agreed, it's not a fatal error

Reply:
I would suggest use pure JSC calls here. `validated_get_function` will throw if this function is not found. 
Then of course `extensions.js` should not set this to a an empty function.  but leave it undefined if this is not found in `JSTimers.js`

Reply:
We are already caching some things here. https://github.com/realm/realm-js/blob/master/src/jsc/jsc_init.cpp#L29
Basically you need to create a protected handle to this function on init time as we do  here https://github.com/realm/realm-js/blob/master/src/jsc/jsc_class.hpp#L60
Then you can call it any time like we do for `Obect.defineProperty` here https://github.com/realm/realm-js/blob/master/src/jsc/jsc_class.hpp#L513

Reply:
That's really helpful, thanks, will take a look later/tomorrow!

Reply:
and you can use `JSValueIsUndefined` to check if this one didn't get a valid function on init